zzz-dynamic-commit.md

```diff
@@ -0,0 +1,217 @@
+# Extension Bolt ZZZ: Dynamic Commitments
+
+Authors:
+  * Olaoluwa Osuntokun <roasbeef@lightning.engineering>
+  * Eugene Siegel <eugene@lightning.engineering>
+
+Created: 2022-06-24
+
+# Table of Contents
+
+- [Introduction](#introduction)
+
+- [Specification](#specification)
+  * [Proposal Messages](#proposal-messages)
+    + [`dyn_begin_propose`](#-dyn_begin_propose-)
+    + [`dyn_propose`](#-dyn_propose-)
+    + [`dyn_propose_reply`](#-dyn_propose_reply-)
+  * [Reestablish](#reestablish)
+    + [`channel_reestablish`](#-channel_reestablish-)
+
+# Introduction
+
+Dynamic commitments are a way to propose changing some static channel parameters that
+are initially set in the `open_channel` and `accept_channel` messages. This document
+outlines how this can be done on-the-fly with new messages.
+
+# Specification
+
+## Proposal Messages
+
+Three new messages are introduced that let each side propose what they want to
+change about the channel.
+
+### `dyn_begin_propose`
```

Comment:
Would it make sense to re-use #869 here? At a glance it can probably be quite trivially expanded to include some of the functionality being used here: 
1. Add a `reject` field to the message to `stfu` message
2. (optionally) Add a `message_type` tlv that indicates why you're pausing the channel (eg, in this example `message_type=115/dyn_propose`)

Reply:
I made it this way because 869 allows HTLC's on the commit tx, and I didn't want to have to think about a new dust limit trimming HTLCs -- but a general flow like 869 could be useful here and probably other future flows

Reply:
> but we need to figure out the warning message bit where it contains a rejected tlv which indicates the fields and why they're rejected (high/low).

I can take a stab at this, looked into it a while back. I think @rustyrussell's idea of bad field / bad value / suggested value will fit pretty well for this use case. Simple enough to repeat that field if we want to point out all the fields we don't like (although I agree that in practise the first one we encounter and don't like should be good enough)

Reply:
cool, feel free to request a review